pandas/core/nanops.py

```diff
@@ -284,14 +286,8 @@ def _get_values(
 
     mask = _maybe_get_mask(values, skipna, mask)
 
-    if is_datetime64tz_dtype(values):
-        # lib.values_from_object returns M8[ns] dtype instead of tz-aware,
-        #  so this case must be handled separately from the rest
-        dtype = values.dtype
-        values = getattr(values, "_values", values)
-    else:
-        values = lib.values_from_object(values)
-        dtype = values.dtype
+    values = extract_array(values, extract_numpy=True)
```

Comment:
According to the typing and docstring, `values` is always an ndarray, so if those are correct, this should not be needed

Reply:
> so if those are correct

they are not.